src/librustdoc/passes/unindent_comments.rs

```diff
@@ -35,65 +34,69 @@ impl clean::Attributes {
 }
 
 fn unindent_fragments(docs: &mut Vec<DocFragment>) {
-    for fragment in docs {
-        fragment.doc = unindent(&fragment.doc);
-    }
-}
-
-fn unindent(s: &str) -> String {
-    let lines = s.lines().collect::<Vec<&str>>();
     let mut saw_first_line = false;
     let mut saw_second_line = false;
-    let min_indent = lines.iter().fold(usize::MAX, |min_indent, line| {
-        // After we see the first non-whitespace line, look at
-        // the line we have. If it is not whitespace, and therefore
-        // part of the first paragraph, then ignore the indentation
-        // level of the first line
-        let ignore_previous_indents =
-            saw_first_line && !saw_second_line && !line.chars().all(|c| c.is_whitespace());
 
-        let min_indent = if ignore_previous_indents { usize::MAX } else { min_indent };
+    if let Some(min_indent) = docs
+        .iter()
+        .map(|fragment| {
+            fragment.doc.lines().fold(usize::MAX, |min_indent, line| {
+                // After we see the first non-whitespace line, look at
+                // the line we have. If it is not whitespace, and therefore
+                // part of the first paragraph, then ignore the indentation
+                // level of the first line
+                let ignore_previous_indents =
+                    saw_first_line && !saw_second_line && !line.chars().all(|c| c.is_whitespace());
 
-        if saw_first_line {
-            saw_second_line = true;
-        }
+                let min_indent = if ignore_previous_indents { usize::MAX } else { min_indent };
 
-        if line.chars().all(|c| c.is_whitespace()) {
-            min_indent
-        } else {
-            saw_first_line = true;
-            let mut whitespace = 0;
-            line.chars().all(|char| {
-                // Compare against either space or tab, ignoring whether they
-                // are mixed or not
-                if char == ' ' || char == '\t' {
-                    whitespace += 1;
-                    true
+                if saw_first_line {
+                    saw_second_line = true;
+                }
+
+                if line.chars().all(|c| c.is_whitespace()) {
+                    min_indent
                 } else {
-                    false
+                    saw_first_line = true;
+                    let mut whitespace = 0;
+                    line.chars().all(|char| {
+                        // Compare against either space or tab, ignoring whether they
+                        // are mixed or not
+                        if char == ' ' || char == '\t' {
+                            whitespace += 1;
+                            true
+                        } else {
+                            false
+                        }
+                    });
```

Comment:
We could probably simplify this a bit.
```suggestion
                    // Compare against either space or tab, ignoring whether they are
                    // mixed or not
                    let whitespace = line.chars().take_while(|c| c == ' ' || c == '\t').count();
```

Reply:
:+1: